ios/MaasCore/Sources/MaasCore/Api/ApiConfiguration.swift

```diff
@@ -1,30 +1,16 @@
-class ApiConfiguration: ApiConfig {
-    
-    private init() {}
+public protocol ApiConfig: class {
+    var baseUrl: String { get }
     
-    static let shared = ApiConfiguration()
-    static var config: CoreBinary.ApiConfiguration { shared.config }
-
-    var baseUrl: String { "" }
-    var apiKey: String { "" }
+    ///  - Tag: ApiConfig.apiKey
+    var apiKey: String { get }
     
-    func getIdToken() -> String? {
-        return ""
-    }
+    ///  - Tag: ApiConfig.getIdToken
+    func getIdToken() -> String?
     
-    var logger: Logger? { ApiLogger.default }
-}
+    /// - Tag: ApiConfig.refreshIdToken
+    ///  must ensure that when completion is called, [getIdToken()](x-source-tag://ApiConfig.getIdToken) will return refreshed token
+    func refreshIdToken(completion: @escaping (String?) -> ())
```

Comment:
One final thought - what do you think about making this method optional? i.e. if the SDK user specifies it, they *opt-in* to automatic retry on unauthorized, otherwise they *opt-out*. I would like the retry functionality we provide to be explicit and optional, if possible, to minimize potentially surprising behavior.

Reply:
Good idea 👍  added a default implementation, to make declaring it optional